doc/man3/BIO_s_datagram.pod

```diff
@@ -0,0 +1,242 @@
+discussion.
+
+=item
+
+Unlike L<BIO_read(3)> and L<BIO_write(3)>, the L<BIO_sendmmsg(3)> and
+L<BIO_recvmmsg(3)> methods are stateless and do not cause the internal state of
+the BIO_s_datagram() to change.
+
+=back
+
+Various controls are available for configuring the BIO_s_datagram():
+
+=over 4
+
+=item BIO_ctrl_dgram_connect (BIO_CTRL_DGRAM_CONNECT)
+
+This is equivalent to calling L<BIO_dgram_set_peer(3)>.
+
+Despite its name, this function does not cause the underlying socket to become
+connected.
+
+=item BIO_ctrl_set_connected (BIO_CTRL_SET_CONNECTED)
+
+This informs the BIO_s_datagram() whether the underlying socket has been
+connected, and therefore how the BIO_s_datagram() should attempt to use the
+socket.
+
+If the peer argument is non-NULL, BIO_s_datagram() assumes that the underlying
+socket has been connected and will attempt to use the socket using OS APIs which
+do not specify peer addresses (for example, send(3) and recv(3) or
+similar). The peer argument should specify the peer address to which the socket
+is connected.
+
+If the peer argument is NULL, BIO_s_datagram() assumes that the underlying
+socket is not connected and will attempt to use the socket using an OS APIs
+which specify peer addresses (for example, sendto(3) and recvfrom(3)).
+
+This control does not affect the operation of L<BIO_sendmmsg(3)> or
+L<BIO_recvmmsg(3)>.
+
+=item BIO_dgram_get_peer (BIO_CTRL_DGRAM_GET_PEER)
+
+This outputs a B<BIO_ADDR> which specifies one of the following values,
+whichever happened most recently:
+
+=over 4
+
+=item
+
+The peer address last passed to BIO_dgram_set_peer(), BIO_ctrl_dgram_connect()
+or BIO_ctrl_set_connected().
+
+=item
+
+The peer address of the datagram last received by a call to BIO_read().
+
+=back
+
+=item BIO_dgram_set_peer (BIO_CTRL_DGRAM_SET_PEER)
+
+Sets the peer address to be used for subsequent writes to this BIO.
+
+Warning: When used with an unconnected network socket, the value set may be
+modified by future calls to L<BIO_read(3)>, making use of BIO_s_datagram()
+hazardous when used with unconnected network sockets; see above.
+
+This does not affect the operation of L<BIO_sendmmsg(3)>, and L<BIO_recvmmsg(3)>
+does not affect this value.
+
+=item BIO_dgram_recv_timeout (BIO_CTRL_DGRAM_GET_RECV_TIMER_EXP)
+
+Returns 1 if the last I/O operation performed on the BIO (for example, via a
+call to L<BIO_read(3)>) may have been caused by a receive timeout.
+
+=item BIO_dgram_send_timedout (BIO_CTRL_DGRAM_GET_SEND_TIMER_EXP)
+
+Returns 1 if the last I/O operation performed on the BIO (for example, via a
+call to L<BIO_write(3)>) may have been caused by a send timeout.
+
+=item BIO_dgram_get_mtu_overhead (BIO_CTRL_DGRAM_GET_MTU_OVERHEAD)
+
+Returns a quantity in bytes which is a rough estimate of the number of bytes of
+overhead which should typically be added to a datagram payload size in order to
+estimate the final L3 packet size. In most cases, the maximum datagram payload
```

Comment:
It's not really clear here what you're trying to say. As far as I know, the UDP overhead 8 bytes, and the minimum IPv4 overhead is 20 bytes. Does this function return 28 in case of an AF_INET socket?

Is this something we use internally?

Reply:
It probably should not say "L3", but write it full, and give IP as an example.

Reply:
>Does this function return 28 in case of an AF_INET socket?

Yes. See `crypto/bio/bss_dgram.c:454`.

Clarified a bit.